### PR TITLE
executor: Add freqs param to confs['cpufreq']

### PR DESCRIPTION
**Pull request by bjackman**
This is to be used for https://github.com/bjackman/lisa/blob/fie-test/tests/eas/load_tracking.py which I will submit as a separate PR once the dependencies are sorted.

I appreciate that there are some piles of documentation for the config dicts piling up here.. I will have to spend some time on it soon.

### CODE REVIEW

File: libs/utils/executor.py [hunk 1]
```diff
@@ -252,6 +252,14 @@ def _setup_cpufreq(self, tc):
 
         self.target.cpufreq.set_all_governors(cpufreq['governor'])
 
+        if 'freqs' in cpufreq:
+            if cpufreq['governor'] != 'userspace':
+                raise ValueError('Must use userspace governor to set CPU freqs')
+            logging.info(r'%14s - CPU frequencies: %s',
```
Comment (derkling) [inline libs/utils/executor.py]: This seems not rebased on the most recent logging refactored code, isn't it?

File: libs/utils/executor.py [hunk 2]
```diff
@@ -252,6 +252,14 @@ def _setup_cpufreq(self, tc):
...
+                    'CPUFreq', str(cpufreq['freqs']))
+            for cpu, freq in cpufreq['freqs'].iteritems():
```
Comment (derkling) [inline libs/utils/executor.py]: Should not we add a check like:
```python
if len(cpufreq['freqs']) < len(self.target.list_online_cpus()):
    self._log.warning('Frequencies not defined for all CPUs')
```

Maybe the warning should be instead a raise ValueError?

Comment (bjackman) [inline libs/utils/executor.py]: I think it's totally valid to not set frequencies for all CPUs - in fact that's how I've used it in my [use case](https://github.com/bjackman/lisa/blob/fie-test/tests/eas/load_tracking.py#L106)

**Comment by bjackman** [comment]
Rebased and fixed logging

File: libs/utils/executor.py (hunk 2 above)
Comment (derkling) [inline libs/utils/executor.py]: Ok, I did not noticed `cpufreq['freqs']` was a map... then that makes much more sense! ;-)

**Pull request merged**